packages/manager/apps/pci-ai-endpoints/src/hooks/token/useDatagridColumns.tsx

```diff
@@ -127,14 +131,14 @@ export const useDatagridColumns = ({
               {
                 id: 0,
                 label: t('ai_endpoints_token_put'),
-                onClick: () => onUpdate(token),
-                disabled: false,
+                onClick: () => userRole === 'ADMIN' && onUpdate(token),
+                disabled: userRole !== 'ADMIN' || undefined,
```

Comment:
If the logic is relevant. (Can't use directly IAM rules)

Maybe deport logic once in hook ?

```ts
const USER_ADMIN_ROLE = 'ADMIN';

const useUserInfos = (): UseUserInfos => {
  const { shell } = useContext(ShellContext);
  const [role, setRole] = useState<string>('');
  useEffect(() => {
    shell.environment.getEnvironment().then((env) => {
      const roles = env.user?.auth?.roles;
      setRole(roles && roles.length > 0 ? roles[0] : '');
    });
  }, [shell]);
  return { role, isAdmin: userRole === USER_ADMIN_ROLE };
};
```

And have :
```suggestion
                disabled: !isAdmin || undefined,
```